File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/general/GeneralSettingsModule.java
package treehou.se.habit.ui.settings.subsettings.general;


import dagger.Module;
import dagger.Provides;
import treehou.se.habit.module.ViewModule;

@Module
public class GeneralSettingsModule extends ViewModule<GeneralSettingsFragment> {

    public GeneralSettingsModule(GeneralSettingsFragment fragment) {
        super(fragment);
    }

    @Provides
    public GeneralSettingsContract.View provideView() {
        return getView();
    }

    @Provides
    public GeneralSettingsContract.Presenter providePresenter(GeneralSettingsPresenter presenter) {
        return presenter;
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/settings/subsettings/general/GeneralSettingsModule.kt
package treehou.se.habit.ui.settings.subsettings.general


import dagger.Module
import dagger.Provides
import treehou.se.habit.module.ViewModule

@Module
class GeneralSettingsModule(fragment: GeneralSettingsFragment) : ViewModule<GeneralSettingsFragment>(fragment) {

    @Provides
    fun provideView(): GeneralSettingsContract.View {
        return view
    }

    @Provides
    fun providePresenter(presenter: GeneralSettingsPresenter): GeneralSettingsContract.Presenter {
        return presenter
    }
}
